agent/wptdriver/wpt_driver_core.cc

```diff
@@ -972,4 +974,21 @@ bool WptDriverCore::NeedsReboot() {
   }
 
   return _exit;
-}
\ No newline at end of file
+}
+
+bool WptDriverCore::NeedsRestart()
+{
+  if (_settings.IsFileOnDiskUpdated()) {
+    TCHAR path[MAX_PATH];
+    GetModuleFileName(NULL, path, MAX_PATH);
+    lstrcpy(PathFindFileName(path), _T("wptdriver.exe"));
+    CString wptdriver_exe;
+    wptdriver_exe.Format(_T("\"%s\" --restart"), path);

```

Comment:
What if the wptdriver.exe was started with other parameters? Do you think it worth doing the extra mile and reuse the wptdriver.exe `argv` when restarting?